app/models/article.rb

```diff
@@ -123,12 +123,12 @@ class Article < ApplicationRecord
     .declare("l_org_vector tsvector; l_user_vector tsvector") do
     <<~SQL
       NEW.reading_list_document :=
-        to_tsvector('simple'::regconfig, unaccent(coalesce(NEW.body_markdown, ''))) ||
-        to_tsvector('simple'::regconfig, unaccent(coalesce(NEW.cached_tag_list, ''))) ||
-        to_tsvector('simple'::regconfig, unaccent(coalesce(NEW.cached_user_name, ''))) ||
-        to_tsvector('simple'::regconfig, unaccent(coalesce(NEW.cached_user_username, ''))) ||
-        to_tsvector('simple'::regconfig, unaccent(coalesce(NEW.title, ''))) ||
-        to_tsvector('simple'::regconfig,
+        setweight(to_tsvector('simple'::regconfig, unaccent(coalesce(NEW.body_markdown, ''))), 'C') ||
+        setweight(to_tsvector('simple'::regconfig, unaccent(coalesce(NEW.cached_tag_list, ''))), 'B') ||
+        setweight(to_tsvector('simple'::regconfig, unaccent(coalesce(NEW.cached_user_name, ''))), 'D') ||
+        setweight(to_tsvector('simple'::regconfig, unaccent(coalesce(NEW.cached_user_username, ''))), 'D') ||
```

Comment:
I realize I didn't phrase this as a question like I intended. Sorry about that. I meant to ask what the purpose of deprioritizing the author's name/username was. 😄

Reply:
I don't have a strong opinion about that but I copied how it was in "Elasticsearch based times": 

https://github.com/forem/forem/blob/b9ecfef149d7b4f4c9c6690862dbcf8ad20801ea/app/services/search/query_builders/feed_content.rb#L5-L15

Here we used to prioritize titles, then tags, then the body and then everything else. 

IIRC it took some trial and error both @mstruve and @atsmith813 to find the right combination at the time.

That said, it doesn't mean that it can't be changed BUT maybe we can rely on search patterns on DEV to see how users search for stuff. 

My impression is that people tend to look for a keyword, which can be something in the title they remember or the name of a technology. If you look at the [top searched keywords for the past 7 days](https://ui.honeycomb.io/thepracticaldev/datasets/rails/result/AcjYL9gUBMm/a/awTp7ZjdqsL/articles-with-search-term) you'll notice this pattern.

I think people just go to the user's profile if they want to see the user's articles.

All of this is going to be improved when we circle back to add additional filters to the search experience.

For this reasons I would keep it as is until we have further evidence (or if this PR doesn't work :D)

Reply:
This is where I think we would really benefit from having more granular search options then we could allow the user to search by author instead of trying to guess that is what they want. I think if its simpler, keeping it as close to as it was in Elasticsearch would be the way to go for now.